README.md

```diff
@@ -1169,6 +1170,45 @@ app.use(UserManagementAddon, {
 } as UserManagementAddonOptions);
 ```
 
+### Configuring roles and permissions
+
+This framework provides support for more granular access control via _roles_ and _permissions_. These layers allow to fine-tune the `@Authorize` decorator to more specific conditions.
+
+In order to enable this feature, you'll need to supply two additional callbacks, called _providers_: `userRolesProvider` and `userPermissionsProvider`. These functions operate with the scope of an `ApplicationInstance` and receive a `User` resource; they must return an array of strings, containing the names of the roles and permissions, respectively.
+
+> 👆️ Depending on your data sources, you might need to define a `Role` and a `Permission` resource.
+
+For example, a role provider could look like this:
+
+`role-provider.ts`
+```ts
+import { ApplicationInstance, User } from "@ebryn/jsonapi-ts";
+
+export default async function roleProvider(this: ApplicationInstance, user: User): Promise<string[]> {
+  const userRoleProcessor = this.processorFor("userRole");
+  
+  return (
+    await roleProcessor.getQuery()
```

Comment:
Everything else it's great, but getQuery is not defined nor documented anywhere (I checked, we have another example that uses it: https://github.com/ebryn/jsonapi-ts/blob/master/README.md#using-computed-properties-in-a-processor ).
I believe that we should explain what does getQuery does, what can be used for, and what's it's output, so that when we use it in examples, the reader has a reference to what it does.

Reply:
I'll add a separate issue for that. It's part of the `KnexProcessor` API.

Reply:
great!